kernel/src/arch/riscv/cpu.rs

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::{
+    cpu::{CpuExceptionInfo, RawGeneralRegs, UserContext},
+    Pod,
+};
+
+use crate::{cpu::LinuxAbi, thread::exception::PageFaultInfo, vm::perms::VmPerms};
+
+impl LinuxAbi for UserContext {
+    fn syscall_num(&self) -> usize {
+        self.a7()
+    }
+
+    fn syscall_ret(&self) -> usize {
+        self.a0()
+    }
+
+    fn set_syscall_ret(&mut self, ret: usize) {
+        self.set_a0(ret)
+    }
+
+    fn syscall_args(&self) -> [usize; 6] {
+        [
+            self.a0(),
+            self.a1(),
+            self.a2(),
+            self.a3(),
+            self.a4(),
+            self.a5(),
+        ]
+    }
+
+    fn set_tls_pointer(&mut self, tls: usize) {
+        self.set_tp(tls);
+    }
+
+    fn tls_pointer(&self) -> usize {
+        self.tp()
+    }
+}
+
+/// General-purpose registers.
+#[derive(Debug, Clone, Copy, Pod, Default)]
+#[repr(C)]
+pub struct GpRegs {
```

Comment:
I'm not sure, but now we already have the `GeneralRegs` in ostd, we may not need this `GpRegs` any more.